### ISSUE TITLE
Add a thesis exposé example

### ISSUE THREAD

**Issue by hochleitner** [Digital-Media/HagenbergThesis#155]
### Discussed in https://github.com/Digital-Media/HagenbergThesis/discussions/153

<div type='discussions-op-text'>

<sup>Originally posted by **hochleitner** January 12, 2023</sup>
In our scientific writing courses in the bachelor programs, students create an exposé for their bachelor's thesis. They'll use the Hagenberg Thesis template to write their thesis, so we also have them write their exposé using it as well.

Since the bachelor's thesis is, for many, the first experience with scientific writing, we give them an exposé example document. That way, they know what's expected of them and what the structure of an exposé looks like. And while working on it, they also become familiar with the template. The document is located here: https://github.com/Digital-Media/MTD248-WA4.

@imagingbook, how do you feel about including this example in Hagenberg Thesis? It could become `HgbThesisExposeDE` (if you don't like the content, we can also come up with something else).

Including it would make it less confusing because, at the moment, there are two repos. Plus, it would get updated with the latest additions from the package automatically. Also, I think that an exposé is something that many students could benefit from.</div>

**Comment by hochleitner** [comment]
Do you think a separate class option is necessary? `expose` (similar to `internship`)? This could create a slightly different cover page with "Exposé" on it as well as something like "proposed supervisor" since this is usually not clear at the time of the writing.

**Comment by imagingbook** [comment]
Yes, and it should take only a few IF's inside the existing macros that generate the front pages.
I would not name it `expose` though for reasons of clarity (it would be _Exposee_ in German and _exposè_ in English). How about simply `proposal`? And I suggest to make it an _extra_ option, such that a document would start like
```
\documentclass[bachelor,proposal,german,smartquotes]{hgbthesis}
```
To subsequently convert to the actual thesis, authors should only need to remove the `proposal` option.

**Comment by hochleitner** [comment]
That is a great idea, let's do it like that.

One other thing I'm not fully sure of is how to deal with the chapters. A proposal in our example is just 5 sections.

Leaving out the chapter command gives 0.1, 0.2 and so on.
Setting one chapter command just for the sake of numbering is also not too elegant (but that's what we did so far).

Switching to an article is very fundamental and would clash with the idea of just removing the proposal option.

I'm sure, you have an elegant solution for this. 😬

**Comment by imagingbook** [comment]
@hochleitner Yes, I wondered about the lonely chapter too.
Let me think about it, I am already playing with the option switches ...

**Comment by imagingbook** [comment]
@hochleitner I made a quick draft for a bachelor thesis proposal, pls. see branch [thesis-proposal-option](https://github.com/Digital-Media/HagenbergThesis/tree/thesis-proposal-option).

- The `proposal` option removes the copyright and declaration (Eidesstattliche ..) pages, which don't seem necessary.
- The document type and advisor entries are modified.
- The (single) chapter and all sections are _starred_, i.e., carry no numbers (and thus labels and cross-references make no sense). There are manual insertions to the TOC, alternatively the TOC could be dropped (automatically).

Let me know what you think...

**Comment by hochleitner** [comment]
I like it but I think it would be nice if we could use \chapter instead of \chapter* (same with the sections). This would mean, we don't need manual TOC entries and could use labels and refs.

My idea was to remove the chapter number if the option proposal is active and sections get the "full" numbers (1, 2, 3, 4, 5). I tried this using titlesec but so far I didn't get it quite right.

Deactivating the TOC could also make sense but this would mean removing the command from main.tex, right? Or wrapping it with an if/else, which is also not very beautiful, so I'd just leave it on.

I renamed the directory btw. An E was missing.

**Comment by hochleitner** [comment]
Okay, I got a working version of what it could look like.

The document now uses `\chapter` and `\section` instead of `\chapter*` and `\section*`; thus, there is no need to add anything manually to the TOC, and we can keep the `\tableofcontents` command.

1. I'm hiding the chapter header ("Kapitel 1") on the first page using a command from `titlesec`, which is already included.
2. To remove the chapter number from the header on the following pages, I updated the `\chaptermark` using `fancyhdr` (also included).
3. To get the number out of the TOC, I'm using `tocbasic` (also included). I define an empty number format, apply it, and then remove the indentation.
4. Finally, I removed the chapter numbers from sections, so 1.1 becomes 1, 1.2 becomes 2, etc. and figures (so we get "Abbildung 1" instead of 1.1.

This way, we should get a cleaner format for the exposé, and if students remove the `proposal` option, they do not need to change the markup of their chapter documents.

Disclaimer: Right now, I added all those changes to `main.tex` so you can see what it looks like. Of course, these have to go into the .sty files in conditionals, so they only apply when the `proposal` option is set.

I updated the branch with these changes, please let me know what you think.

**Comment by imagingbook** [comment]
Very slick indeed! I am currently merging everything into the class file and fixing it up. PR coming soon ...

**Comment by hochleitner** [comment]
Thanks. I just remembered, I didn't test tables or code. These will probably have their chapter numbers removed as well. Could you maybe test that in the process. These might be things needed in a proposal. Maybe.

**Comment by imagingbook** [comment]
Yes, these have been fixed too.

**Comment by hochleitner** [comment]
Oh, and one other thing: could you make the proposal type available for master's theses as well? I talked to the IM guys, they are thinking about moving away from the thesis application form to a separate exposé, so they might need that as well.

**Comment by imagingbook** [comment]
The new `proposal` option can now be used on `bachelor` and `master` type documents and should have no effect when used in others. Updated the manual, tutorial and `README.md` to reflect the changes.
I also added (in Tutorial Sec. 4.4.5) a note on the [_Ipe Extensible Drawing Editor_](https://ipe.otfried.org/), which is an attractive option to produce graphics with native embedded LaTeX typography.
